pkg/kube/cluster-init.sh

```diff
@@ -413,18 +439,44 @@ get_enc_status() {
     fi
 }
 
+
 # When transitioning from single node to cluster mode, need change the controller
 # provided token for the cluster
+
+rotate_cluster_token() {
+        local token="$1"
+        /usr/bin/k3s token rotate --new-token "$token"
+        local status=$?
+        if [ $status -ne 0 ]; then
+                logmsg "Failed to rotate token. Exit status: $status"
+        else
+                logmsg "Token rotated successfully."
+        fi
+        return $status
+}
+
 change_to_new_token() {
   if [ -n "$cluster_token" ]; then
-    /usr/bin/k3s token rotate --new-token "$cluster_token"
+    logmsg "Rotate cluster token size: ${#cluster_token}"
+    rotate_cluster_token "$cluster_token"
+    # Set the starttime before entering the while loop
+    starttime=$(date +%s)
+
     while true; do
         if grep -q "server:$cluster_token" /var/lib/rancher/k3s/server/token; then
```

Comment:
If the goal is to monitor changes on `/var/lib/rancher/k3s/server/token`, you can improve this while by using `inotifyd` to catch changes in the file instead of sleep every 2 or 5 seconds...